src/core/session.ts

```diff
@@ -305,6 +305,13 @@ export class Session
     this.notifyApplicationAfterFrameRender(fetchResponse, frame)
   }
 
+  async frameMissing(frame: FrameElement, fetchResponse: FetchResponse): Promise<void> {
+    const responseHTML = await fetchResponse.responseHTML
```

Comment:
Since servers (like Rails) can [respond to Frame requests with incomplete HTML documents](https://github.com/hotwired/turbo-rails/pull/232), I'm unsure about how to best handle a response without a matching frame.

I'm not sure whether or not Drive should re-use the response HTML or issue a new request to fetch a _full_ page of content. My gut tells me to always make a full request, but I'm curious if there's a quick win I'm not considering.

Reply:
I think the current default is correct, because the majority case here will be to deal with pages like 500 errors or authentication expired errors, where full pages will be returned. But I think we can help developers here by logging a warning to point them in the direction of what's going on.

Reply:
Ah, I see you changed it to a full visit instead. I guess it doesn't matter that much. This behavior is for the exceptional case in any circumstance. Not for the normal, steady flow.